### PR TITLE
Moved DeepCopyPointer.h to DataFormats/Common

### PR DESCRIPTION
**Pull request by Teemperor**
This is supposed to fix the cycle: DataFormats/GeometryCommonDetAlgo → DataFormats/GeometrySurface → DataFormats/GeometryCommonDetAlgo

Open for alternatives to doing this.

More information here: https://teemperor.de/pub/cms/cycle1.cycle.html

### CODE REVIEW

**Comment by Teemperor** [comment]
see https://github.com/cms-sw/cmssw/issues/15248 for more information

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by Dr15Jones** [comment]
I am not happy with continually putting items in to DataFormats/Common. These items in question are not ones the framework wants to be responsible for.

**Comment by Dr15Jones** [comment]
-1

**Comment by Dr15Jones** [comment]
How about forming a `DataFormats/GeometryCommon` package to hold all the classes which are shared between those two packages?

**Comment by slava77** [comment]
just a note from reco perspective: we are waiting for the resolution to the "-1" from core.

**Comment by slava77** [comment]
```git grep -w DeepCopyPointer ``` does not show use cases, only the definition in DeepCopyPointer.h and the includes, sowe of which apparently trigger the cycle.
There is only DeepCopyPointer in DeepCopyPointer.h.
So, it seems like the solution is to remove the include

File: DataFormats/GeometryCommonDetAlgo/interface/GlobalErrorBase.h [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #ifndef GlobalErrorType_H
 #define GlobalErrorType_H
 
-#include "DataFormats/GeometryCommonDetAlgo/interface/DeepCopyPointer.h"
+#include "DataFormats/Common/interface/DeepCopyPointer.h"
```
Comment (slava77) [inline DataFormats/GeometryCommonDetAlgo/interface/GlobalErrorBase.h]: as commented in the main thread, I think that this include is superfluous (same in all other cases)

**Comment by VinInn** [comment]
+1 for nuke it

**Comment by slava77** [comment]
@Teemperor
ping
please clarify on the status of resolution of https://github.com/cms-sw/cmssw/pull/21555#pullrequestreview-81625949
Thank you.

**Pull request closed without merging**